packages/lesswrong/server/repos/VotesRepo.ts

```diff
@@ -109,4 +117,20 @@ export default class VotesRepo extends AbstractRepo<DbVote> {
       WHERE ARRAY_POSITION("authorIds", $1) IS NOT NULL
     `, [oldUserId, newUserId]);
   }
+  
+  async getDigestPlannerVotesForPosts(postIds: string[]): Promise<Array<PostVoteCounts>> {
+    return await logIfSlow(async () => await this.getRawDb().manyOrNone(`
+      SELECT p._id as "postId",
+        count(v._id) FILTER(WHERE v."voteType" = 'smallUpvote') as "smallUpvoteCount",
+        count(v._id) FILTER(WHERE v."voteType" = 'bigUpvote') as "bigUpvoteCount",
+        count(v._id) FILTER(WHERE v."voteType" = 'smallDownvote') as "smallDownvoteCount",
+        count(v._id) FILTER(WHERE v."voteType" = 'bigDownvote') as "bigDownvoteCount"
+      FROM "Posts" p
+      JOIN "Votes" v tablesample system(50) ON v."documentId" = p."_id"
+      WHERE p._id = ANY($1)
```

Comment:
IIRC, `ANY` is really bad at using indexes. This is the primary key so it might be ok, but I'm wary of this because this is exactly the same thing that brought the site down when we migrated client ids to postgres.

You're probably better off using either `p._id IN ($1:csv)` or `$1 @> ARRAY[p._id]`.

Reply:
Appreciate this, thanks!